src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1560,15 +1569,43 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             _terminal->SetBlockSelection(false);
             _renderer->TriggerSelection();
             _UpdateSelectionMarkersHandlers(*this, winrt::make<implementation::UpdateSelectionMarkersEventArgs>(true));
+
+            foundResults->TotalMatches(gsl::narrow<int32_t>(_searcher.Results().size()));
+            foundResults->CurrentMatch(gsl::narrow<int32_t>(_searcher.CurrentMatch()));
+
+            _terminal->AlwaysNotifyOnBufferRotation(true);
         }
 
         // Raise a FoundMatch event, which the control will use to notify
         // narrator if there was any results in the buffer
-        _FoundMatchHandlers(*this, winrt::make<implementation::FoundResultsArgs>(foundMatch));
+        _FoundMatchHandlers(*this, *foundResults);
+    }
+
+    Windows::Foundation::Collections::IVector<int32_t> ControlCore::SearchResultRows()
+    {
+        auto lock = _terminal->LockForWriting();
+        _searcher.ResetIfStale(*GetRenderData());
+
+        auto results = std::vector<int32_t>();
+
+        // use a map to remove duplicates
+        std::map<int32_t, bool> rows;

```

Comment:
I don't think this map is necessary. Since returning redundant Y coordinates is fine-ish we can do something like this instead:
```cpp
auto lastRow = til::CoordTypeMin;
for (const auto& match : _searcher.Results())
{
    if (match.start.y > lastRow)
    {
        results.push_back(row);
        lastRow = match.start.y
    }
}
```

If we wanted perfectly unique entries then I would personally use:
```cpp
auto lastRow = til::CoordTypeMin;
for (const auto& match : _searcher.Results())
{
    if (match.start.y != lastRow)
    {
        results.push_back(row);
        lastRow = match.start.y
    }
}

std::sort(results.begin(), results.end());
results.erase(std::unique(results.begin(), results.end()), results.end());
```

But I think the former is fine right?

Reply:
Oh that's a great idea, thanks!